lib/SPIRVProducerPass.cpp

```diff
@@ -90,6 +90,28 @@ const glsl::ExtInst kGlslExtInstBad = static_cast<glsl::ExtInst>(0);
 
 const char *kCompositeConstructFunctionPrefix = "clspv.composite_construct.";
 
+// SPIRV Module Sections (per 2.4 of the SPIRV spec)
+// These are used to collect SPIRVInstructions by type on-the-fly.
+enum SPIRVSection {
+  kCapabilities,
+  kExtensions,
+  kImports,
+  kMemoryModel,
+  kEntryPoints,
+  kExecutionModes,
+
+  kDebug,
+  kAnnotations,
+
+  kTypes,
```

Comment:
This shouldn't be broken up like this. Types can depend on constants (e.g. array sizes). Forcing this extra separation will lead to invalid IR. Types, constants and global variables can all be intermixed.

Reply:
Agreed. In a future CL I set kConstants = kTypes. That way we can still collect by type but will be the same bucket.

Reply:
I can make the change for this CL.